src/dotnet/commands/RestoringCommand.cs

```diff
@@ -20,7 +20,8 @@ private IEnumerable<string> ArgsToForwardToRestore()
                 !a.StartsWith("/t:") &&
                 !a.StartsWith("/target:") &&
                 !a.StartsWith("/ConsoleLoggerParameters:") &&
-                !a.StartsWith("/clp:"));
+                !a.StartsWith("/clp:") &&
+                !a.StartsWith("/p:TargetFramework"));
```

Comment:
I would vastly prefer that we actually implement what we designed: that only the implicit msbuild args are handled specially. This is just approximating that and it's simply incorrect in the "corner cases".